### PR TITLE
[PWD-2508] - Finalize Voucher Page (Transaction Journal)

### PR DESCRIPTION
**Pull request by katherinedavenia**
### What type of PR is this? (_check all applicable_)

- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update
- [ ] Setup
- [ ] Code Refactor
- [ ] Performance Improvements
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Revert

### CODE REVIEW

File: src/views/VoucherView/components/TransactionHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+  const groupedData: Record<string, CorporateTokenJournal[]> = {}
+
+  tokenJournals.forEach((tokenJournal) => {
+    const createdAt = format(new Date(tokenJournal.createdAt), 'dd MMMM yyyy')
+    if (!groupedData[createdAt]) {
+      groupedData[createdAt] = []
+    }
+    groupedData[createdAt].push(tokenJournal)
+  })
```
Comment (shadiqfauzi) [inline src/views/VoucherView/components/TransactionHistory.tsx:72]: ini mungkin bisa pakai useMemo biar ga di compute tiap render

File: src/views/VoucherView/components/TransactionHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+              if (corporateTokenProgramSales[0]) {
+                return (
+                  <ProgramSalesJournal sales={corporateTokenProgramSales[0]} />
+                )
+              }
+              if (corporateTokenServiceSales[0]) {
+                return (
+                  <ServiceSalesJournal sales={corporateTokenServiceSales[0]} />
+                )
+              }
```
Comment (shadiqfauzi) [inline src/views/VoucherView/components/TransactionHistory.tsx:99]:
```suggestion
              if (corporateTokenServiceSales[0]) {
                return (
                  <ServiceSalesJournal sales={corporateTokenServiceSales[0]} />
                )
              }
              return null
```
jaga2 aja wkwk

File: src/views/VoucherView/components/TransactionHistoryModal.tsx [hunk 3]
```diff
@@ -201,7 +179,7 @@ const HistoryModal = ({
                 <SectionTitle>Rincian Pesanan</SectionTitle>
                 <Box mt={2}>
                   <Paragraph2 color="neutral100.main">
-                    {type} {programName}
+                    {/* {type} {programName} */}
```
Comment (Vesals) [inline src/views/VoucherView/components/TransactionHistoryModal.tsx:182]: kelupaan ??

Comment (ansorudin) [reply src/views/VoucherView/components/TransactionHistoryModal.tsx:182]: ouch iya wkwkwk

**Review by Vesals — APPROVED** [review]
🚀

**Review by shadiqfauzi — DISMISSED** [review]
okeh gaskeunn

**Review by haikal1223 — DISMISSED** [review]
TatanBabyTatanGIF

**Review by Vesals — APPROVED** [review]

**Review by haikal1223 — APPROVED** [review]

**Pull request merged**